src/hotspot/share/opto/superword.cpp

```diff
@@ -1691,42 +1743,52 @@ void SuperWord::filter_packs_for_alignment() {
   AlignmentSolution const* current = new TrivialAlignmentSolution();
   int mem_ops_count = 0;
   int mem_ops_rejected = 0;
+  int new_packset_length = 0;
   for (int i = 0; i < _packset.length(); i++) {
-    Node_List* p = _packset.at(i);
-    if (p != nullptr) {
```

Comment:
Note: mostly change due to indentation.
Rename `p -> pack`
And added the `new_packset_length`, `keep` logic, instead of
`_packset.at_put(i, nullptr);` and then later removing all `nullptr` with `compress_packset`.